ERCS/erc-7730.md

```diff
@@ -0,0 +1,1033 @@
+| `123456789A`    | Field Value = Value ['12','34','56','78','9a'] |
+
+---
+| **`calldata`** | |
+| --- | --- |
+| *Description*   | Data contains a call to another smartcontract. To look for relevant ERC-7730 files matching this embedded calldata, use `callee` parameter and `selector`. If an ERC-7730 is not found or if embedded calldata are not supported by the wallet, it MAY display a hash of the embedded calldata instead, with target `calleePath` resolved to a trusted name if possible.  |
+| *Parameters*    | ---  |
+| `selector`      | Optional selector, if not present the first 4 bytes of the calldata are interpreted as the selector |
+| `calleePath`    | A path reference to the contract being called |
+| *Examples*      |  |
+
+#### Address
+
+Formats useable for address
+
+---
+| **`raw`** | |
+| --- | --- |
+| *Description*   | Display address as an [EIP-55](./eip-55.md) formatted string. Truncation is device dependent  |
+| *Parameters*    | None  |
+| *Examples*      | --- |
+| `0x5aAe...eAed`       | Field Value 0x5aAeb6053F3E94C9b9A09f33669435E7Ef1BeAed  |
+
+---
+| **`addressName`** | |
+| --- | --- |
+| *Description*   | Display address as a trusted name if a trusted source exists, an EIP-55 formatted address otherwise. See [next section](#address-trusted-sources) for a reference of trusted sources |
+| *Parameters*    | ---  |
+| `type`          | The expected type of the address. If set, the wallet SHOULD restrict name lookup to relevant sources, and SHOULD check that the address matches the type  |
+| `source`        | See [next section](#address-trusted-sources) for a reference of trusted sources |
+| *Examples*      | --- |
+| `vitalik.eth`      | Field Value 0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045  |
+
+#### Address trusted sources
+
+Address names trusted sources specify which type and source of trusted names SHOULD be used to replace an address with a human readable names. 
+
+When specified a wallet MUST only use specified sources to resolve address names. Wallet MUST verify the type of the address if able to. 
+
+When omitted, a wallet MAY use any source to resolve an address.
+
+| Address type  | Description |
+| ---           | --- |
+| wallet        | Address is an account controlled by the wallet |
+| eoa           | Address is an Externally Owned Account         |
+| contract      | Address is a well known smartcontract  |
+| token         | Address is a well known ERC-20 token  |
+| collection    | Address is a well known NFT collection  |
+
+A wallet MAY verify that a `wallet` address is in fact controlled by the wallet, and reject signing if not the case.
+
+| Source type   | Description |
+| ---           | --- |
+| local         | Address MAY be replaced with a local name trusted by user. Wallets MAY consider that `local` setting for `sources` is always valid |
+| ens           | Address MAY be replaced with an associated ENS domain |
+
+### Wallets
+
+#### Ledger
+
+[TBD] Link to ledger specification
```

Comment:
```suggestion
<!-- TBD --> Link to ledger specification
```

Just a note, most external links aren't allowed. If you believe a particular origin should be allowed, please see [EIP-5757](https://eips.ethereum.org/EIPS/eip-5757).